src/platform/linux/osfiles.cpp

```diff
@@ -47,13 +49,62 @@
 #include <unistd.h>
 
 #include <iostream>
+#include <fstream>
+// #include <sstream>
+#include <regex>
+#include <dirent.h>
+
+std::string dll_search_path;
+
+// NAN_METHOD is a macro; it's shorthand so you don't have to write that the
+// first and only parameter is of type Nan::FunctionCallbackInfo<v8::Value>
+NAN_METHOD(OSFilesSetDllSearchPath)
+{
+    // Parse parameter from the FunctionCallbackInfo received, convert into a std::string
+    if (info.Length() > 0) {
+        if (info[0]->IsString()) {
+            v8::String::Utf8Value param1(info[0]->ToString());
+            std::string path = std::string(*param1);
+
+            printf("\nlinux/osfiles.cpp: SetDllSearchPath() called with %s\n\n",
+                path.c_str()
+            );
+
+            dll_search_path.assign(path);
+        }
+    }
+    /// TODO: Add some error throwing if no parameters or the parameter is not a string
+}
+
+
+
+// C++ version of ANSI C readlink, as per
+// https://stackoverflow.com/questions/5525668/how-to-implement-readlink-to-find-the-path#5525712
+std::string cpp_readlink(std::string const& path) {
```

Comment:
TODO: remove. this is legacy form a previous implementation (the /proc/self/maps one)